lib/state/ui/middleware.ts

```diff
@@ -8,7 +8,9 @@ let searchTimeout: NodeJS.Timeout;
 
 export const middleware: S.Middleware = store => {
   const updateNotes = () =>
-    store.dispatch(filterAction(filterNotes(store.getState())));
+    store.dispatch(
+      filterAction(filterNotes(store.getState()), store.getState().ui.noteIndex)
```

Comment:
What was it needed for here?

If we change the `note` reducer to return `null` in cases where the note is no longer in the filtered list (instead of using `action.notes[0]`) then could we eliminate passing it here? Whereas the `<NoteList>` could say "oh, I have a `null` note but I have a `previousIndex`" and then keep that logic as a render-tie concern?

Reply:
`FILTER_NOTES` gets run on most actions and will clobber any attempt to set the selected note with notes[0] if none is selected. https://github.com/Automattic/simplenote-electron/pull/1919/files#diff-dfb3add3b5c868976c680f25330e1c32L191

If we remove this functionality from the note reducer for `FILTER_NOTES` we lose note selection on search when a note no longer exists in the results.

Reply:
I have to wonder if it will easier to sort this out sometime down the road when we have more control in the Redux state.